### ISSUE TITLE
Entirely resetting a scene

### ISSUE THREAD

**Issue by LaP0573** [photonstorm/phaser#3717]
We are trying to totally reset a Scene, but it doesn't seem to be working. Basically, we have a scene on which we spawn Graphics (players, landscape and so on). When the player dies and restarts, we want to fully clean the Scene and restart it from scratch (so we want all the Graphics to be removed from the scene, all the fields of the scene to be reverted to their default state, etc). Basically we want to delete it and re-add it (so we go through the constructor again and so on).

How can this be done ?

Just calling:

    scene.stop(OurScene.KEY);
    scene.launch(OurScene.KEY);

Doesn't reset anything, all our fields are still filled with our spawned Graphics (we have Maps filled with Graphics objects). Sidenote : we are using launch() because we run two parallel scenes and need to pass data to scene via launch, I just omitted it from this sample code.

We also tried:

    scene.remove(OurScene.KEY);
    scene.launch(OurScene.KEY);

But this leads to a black screen, and we don't go through the scene's methods anymore. I guess remove fully removes the scene from the whole scene system and launch fails silently, because the scene is no longer known by phaser ?

We also tried:

    scene.remove(OurScene.KEY);
    scene.add(OurScene.KEY, {}, false);
    scene.launch(OurScene.KEY);

Same problem.

So basically, here's the question : how can we totally delete a scene and re-create it from scratch, so that all the fields are back to their starting value, all the graphics are removed from the scene and so on ?

Thanks

**Comment by photonstorm** [comment]
`launch` will start a Scene running in _parallel_ to the current ones. You should be calling `scene.restart` if restarting the same Scene. And you absolutely need to listen for the Scene shutdown event and then clear out anything you've created yourself. Even then, `restart` won't call the Scene constructor again. It will just restart the Scene, which is the same as shutting it down then starting it again, not creating it from scratch.

I would suggest you destroy the current Scene, then start a brand new one, using a brand new key. That is the only way it will run the constructor again. Would be easiest to make your old Scene start the new one, then in the new Scene it destroys the old one.

**Comment by LaP0573** [comment]
Alright I see, thanks a lot for your answer !

**Comment by dondon67** [comment]
On May 31, 2018 1:37 AM, "LaP0573" <notifications@github.com> wrote:

Closed #3717 <https://github.com/photonstorm/phaser/issues/3717>.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/photonstorm/phaser/issues/3717#event-1655238658>, or mute
the thread
